packages/fxa-shared/lib/logger.ts

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Logging module that performs proper checks and string
+ * interpolation before logging message.
+ *
+ */
+
+// tslint:disable-next-line: no-var-requires
+require('jsdom-global')();
+
+interface Logger {
+  _window: Window;
+  info: (...args: any) => void;
+  trace: (...args: any) => void;
+  warn: (...args: any) => void;
+  error: (...args: any) => void;
+}
+
+interface ModuleError {
+  errorModule: {
+    toInterpolatedMessage: (arg0: ModuleError) => string;
+  };
+}
+
+/**
+ * Constructor of log module.
+ *
+ * @param {Object} logWindow Window object that contains console used for logging.
+ *
+ * @constructor
+ */
+const Logger = (function(this: Logger, logWindow?: Window) {
+  this._window = logWindow || window;
+} as any) as new (logWindow?: Window) => Logger;
```

Comment:
Open to suggestions here (and the one in sentry.ts). I got to this because VS Code was complaining:

> 'new' expression, whose target lacks a construct signature, implicitly has an 'any' type.

And I also had to bind `this` to its own type.

Reply:
WFM